src/Hl7.Fhir.ElementModel/TypedElementOnSourceNode.cs

```diff
@@ -74,6 +77,7 @@ private TypedElementOnSourceNode(TypedElementOnSourceNode parent, ISourceNode so
             Definition = definition;
             InstanceType = instanceType;
             _settings = parent._settings;
+            _value = new Lazy<object>(valueFactory);
```

Comment:
This will cause every source node to have a Lazy<T>() allocated, even if the .Value property is never called. Can we use https://docs.microsoft.com/en-us/dotnet/api/system.threading.lazyinitializer.ensureinitialized?view=net-5.0 instead?